src/clj_telnet/core.clj

```diff
@@ -12,6 +13,8 @@
   [^TelnetClient telnet-client]
   (.disconnect telnet-client))
 
+;; TODO: Should set an input data buffer, or an SocketTimeoutException exception will be raised
+;; when some data not read.
```

Comment:
This is a good point. I think I saw this tonight while playing with it. I'll have to try to reproduce it.

I've thought about re-working the read mechanism with core.async. It would constantly read from the client, place it on a channel, and then you'd decide how much you want to read at a time. But it'd constantly be reading. More like what a cli telnet client does.